src/NzbDrone.Core/Notifications/Trakt/TraktService.cs

```diff
@@ -0,0 +1,275 @@
+                    break;
+                case 720:
+                    traktResolution = "hd_720p";
+                    break;
+                case 576:
+                    traktResolution = $"sd_576{scanIdentifier}";
+                    break;
+                case 480:
+                    traktResolution = $"sd_480{scanIdentifier}";
+                    break;
+            }
+
+            return traktResolution;
+        }
+
+        private string MapAudio(EpisodeFile episodeFile)
+        {
+            var traktAudioFormat = string.Empty;
+
+            var audioCodec = episodeFile.MediaInfo != null ? MediaInfoFormatter.FormatAudioCodec(episodeFile.MediaInfo, episodeFile.SceneName) : string.Empty;
+
+            switch (audioCodec)
+            {
+                case "AC3":
+                    traktAudioFormat = "dolby_digital";
+                    break;
+                case "EAC3":
+                    traktAudioFormat = "dolby_digital_plus";
+                    break;
+                case "TrueHD":
+                    traktAudioFormat = "dolby_truehd";
+                    break;
+                case "EAC3 Atmos":
+                case "TrueHD Atmos":
+                    traktAudioFormat = "dolby_atmos";
+                    break;
+                case "DTS":
+                case "DTS-ES":
+                    traktAudioFormat = "dts";
+                    break;
+                case "DTS-HD MA":
+                    traktAudioFormat = "dts_ma";
+                    break;
+                case "DTS-HD HRA":
+                    traktAudioFormat = "dts_hr";
+                    break;
+                case "DTS-X":
+                    traktAudioFormat = "dts_x";
+                    break;
+                case "MP3":
+                case "MP2":
+                    traktAudioFormat = "mp3";
+                    break;
+                case "Vorbis":
+                    traktAudioFormat = "ogg";
+                    break;
+                case "WMA":
+                    traktAudioFormat = "wma";
+                    break;
+                case "AAC":
+                    traktAudioFormat = "aac";
+                    break;
+                case "PCM":
+                    traktAudioFormat = "lpcm";
+                    break;
+                case "FLAC":
+                    traktAudioFormat = "flac";
+                    break;
+                case "Opus":
+                    traktAudioFormat = "ogg";
+                    break;
+            }
+
+            return traktAudioFormat;
+        }
+
+        private string MapAudioChannels(EpisodeFile episodeFile, string audioFormat)
+        {
+            var audioChannels = episodeFile.MediaInfo != null ? MediaInfoFormatter.FormatAudioChannels(episodeFile.MediaInfo).ToString("0.0") : string.Empty;
+
+            // Map cases where Sonarr doesn't handle MI correctly, can purge once mediainfo handling is improved
```

Comment:
`MediaInfoFormatter.FormatAudioChannels` does this based on actual audio channel positions, we should use that here as well, if it's incorrect then we should enhance that there.

Reply:
Yes, Trakt added the below formats

`dolby_digital_plus_atmos` for EAC3 Atmos tracks
`mp2` for MP2 tracks
`ogg_opus` for OPUS tracks.

I already discussed the TV option with him, but he's still debating it as he feels Digital works. He said to give him some time to think about it.

There are some more complex bugs for MediaInfo AudioChannels. 

MediaInfo does not give channel info (just says "object based") for DTS-X tracks. See MediaInfo issue here https://github.com/MediaArea/MediaInfo/issues/299. These are parsed as 8.0 by the Arrs because of the lack of actual channel layout, but need to be sent as 7.1 to Trakt. Not sure if you'd want to bypass the parsing in the Formatter itself.

For 6.0 cases, it is due to the issue we were discussing in Radarr. You can see the preliminary discussion below, and I can give a detailed breakdown of my Radarr files showing what Radarr is parsing vs what MediaInfo is displaying. https://github.com/Radarr/Radarr/issues/4852

For 0,0 cases, I can give some examples as well. These seem to be PCM 1.0 tracks that are being parsed as 0.0 for an unknown reason. 0.0 is not a realistic channel layout (means the file is completely silent).

Reply:
> MediaInfo does not give channel info (just says "object based") for DTS-X tracks.

I don't see a reason to special case that here and not fix it where we format all audio tracks, then we display the correct thing in the UI as well.

The 6.0 cases are definitely special, but honestly I'm inclined to say those stay as-is until things are parsed correctly.

The 0.0 case would be a fallback after formatting.

Reply:
@markus101 Can we not just leave the temporary fix in until MediaInfo fixes it. Maybe just open a issue on Sonarr side that references the open Radarr issue https://github.com/Radarr/Radarr/issues/4852 and any future Radarr fix can then be ported over to Sonarr? I think there are already plans to get this fixed, either by other Radarr folks and/or mediaInfo fixes...

Please advise what I should do for this PR. 

Also regarding the TRAKT TV Option --> @rg9400 and I will keep on top of that and fix it should Trakt do anything about it.

Reply:
At the very least this logic should be ported to the formatter, it’s trying to do the same thing this is, but in a different spot.

Reply:
I'll look into moving the logic soon.

Reply:
What's left to do here?

Reply:
It looks like we are waiting for the following 2 items:
When Radarr/Radarr#5404 is fixed; we'll pull it's corresponding PR as well as Radarr/Radarr#5352